Gems/EMotionFX/Code/Source/Editor/Plugins/Ragdoll/RagdollColliderTranslationManipulators.cpp

```diff
@@ -0,0 +1,193 @@
+    {
+        if (!m_ragdollManipulatorData.m_valid || !m_ragdollManipulatorData.m_colliderNodeConfiguration ||
+            m_ragdollManipulatorData.m_colliderNodeConfiguration->m_shapes.empty())
+        {
+            return;
+        }
+
+        m_translationManipulators.SetLocalPosition(m_ragdollManipulatorData.m_colliderNodeConfiguration->m_shapes[0].first->m_position);
+    }
+
+    void RagdollColliderTranslationManipulators::Teardown()
+    {
+        m_translationManipulators.Unregister();
+    }
+
+    void RagdollColliderTranslationManipulators::ResetValues()
+    {
+        m_ragdollManipulatorData.m_colliderNodeConfiguration->m_shapes[0].first->m_position = AZ::Vector3::CreateZero();
+        m_translationManipulators.SetLocalPosition(AZ::Vector3::CreateZero());
+    }
+
+    AZ::Vector3 RagdollColliderTranslationManipulators::GetPosition(const AZ::Vector3& startPosition, const AZ::Vector3& offset)
+    {
+        const float scale = AZ::GetMax(AZ::MinTransformScale, m_ragdollManipulatorData.m_nodeWorldTransform.GetUniformScale());
+        return startPosition + offset / scale;
+    }
+
+    void RagdollColliderTranslationManipulators::OnManipulatorMoved(const AZ::Vector3& startPosition, const AZ::Vector3& offset)
+    {
+        AZ::Vector3 newPosition = GetPosition(startPosition, offset);
+        m_ragdollManipulatorData.m_colliderNodeConfiguration->m_shapes[0].first->m_position = newPosition;
+        m_translationManipulators.SetLocalPosition(newPosition);
+        AZ::SerializeContext* serializeContext = nullptr;
+        AZ::ComponentApplicationBus::BroadcastResult(serializeContext, &AZ::ComponentApplicationBus::Events::GetSerializeContext);
+        if (serializeContext)
+        {
+            m_ragdollManipulatorData.m_collidersWidget->Update();
+        }
+    }
+
+    void RagdollColliderTranslationManipulators::BeginEditing(const AZ::Vector3& startPosition, const AZ::Vector3& offset)
+    {
+        if (!m_commandGroup.IsEmpty())
+        {
+            return;
+        }
+        m_commandGroup.SetGroupName("Adjust collider");
+
+        const AZ::u32 actorId = m_ragdollManipulatorData.m_actor->GetID();
+        const AZStd::string& nodeName = m_ragdollManipulatorData.m_node->GetNameString();
+        auto colliderType = PhysicsSetup::ColliderConfigType::Ragdoll;
+        const size_t colliderIndex = 0;
+        CommandAdjustCollider* command = aznew CommandAdjustCollider(actorId, nodeName, colliderType, colliderIndex);
+        m_commandGroup.AddCommand(command);
+        command->SetOldPosition(GetPosition(startPosition, offset));
+    }
+
+    void RagdollColliderTranslationManipulators::FinishEditing(const AZ::Vector3& startPosition, const AZ::Vector3& offset)
+    {
+        if (m_commandGroup.IsEmpty())
+        {
+            return;
+        }
+
+        CommandAdjustCollider* command = static_cast<CommandAdjustCollider*>(m_commandGroup.GetCommand(0));
```

Comment:
suggestion: Should this actually be a `azdynamic_cast` (or do a check beforehand to make sure the types are correct?)

Reply:
Switched to azdynamic_cast.